command_test.go

```diff
@@ -2921,3 +2921,30 @@ func TestUnknownFlagShouldReturnSameErrorRegardlessOfArgPosition(t *testing.T) {
 		})
 	}
 }
+
+func TestHelpFuncExecuted(t *testing.T) {
+	helpText := "Long description"
+
+	child := &Command{Use: "child", Run: emptyRun}
+	child.SetHelpFunc(func(cmd *Command, args []string) {
+		_, err := cmd.OutOrStdout().Write([]byte(helpText))
+		if err != nil {
+			t.Error(err)
+		}
+
+		// Test for https://github.com/spf13/cobra/issues/2240
+		if cmd.Context() == nil {
```

Comment:
This looks much better, thanks

 I’ve been thinking about what we’re trying to achieve with this PR. I think we wanna make sure that the correct context can be retrieved by the help function. Right?

Therefore, I was thinking that in this test we should set a context on the root command and then instead of checking for nil here, we should check that we got the same context than the one we created. 

What do you think @Frassle ?

Reply:
Yes that seems reasonable to check in the test.